pkg/kubelet/dockertools/manager.go

```diff
@@ -276,6 +276,38 @@ type containerStatusResult struct {
 	err    error
 }
 
+// determineInfraContainerIP inspects the given container and determines the IP address of that
+// container.  The container is assumed to be the pod infra container; it is the responsibility
+// of the caller to ensure that the container ID passed is that of the infra container.
+func (dm *DockerManager) determineInfraContainerIP(dockerID string, pod *api.Pod) (string, error) {
+	inspectResult, err := dm.client.InspectContainer(dockerID)
+	if err != nil {
+		return "", err
+	}
+	if inspectResult == nil {
+		return "", nil // TODO: should be an error, probably
+	}
+
+	result := ""
+
+	if inspectResult.State.Running {
+		if inspectResult.NetworkSettings != nil {
+			result = inspectResult.NetworkSettings.IPAddress
+		}
+
+		if dm.networkPlugin.Name() != network.DefaultPluginName {
+			netStatus, err := dm.networkPlugin.Status(pod.Namespace, pod.Name, kubeletTypes.DockerID(dockerID))
+			if err != nil {
+				glog.Errorf("NetworkPlugin %s failed on the status hook for pod '%s' - %v", dm.networkPlugin.Name(), pod.Name, err)
```

Comment:
I'm wondering to myself if this should be an error in this and `inspectContainer`, which I cribbed this code from.